src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_jni.c

```diff
@@ -365,49 +376,68 @@ jmethodID g_IteratorNext;
 jclass    g_ListClass;
 jmethodID g_ListGet;
 
+// javax/net/ssl/HostnameVerifier
+jclass    g_HostnameVerifier;
+jmethodID g_HostnameVerifierVerify;
+
+// javax/net/ssl/HttpsURLConnection
+jclass    g_HttpsURLConnection;
+jmethodID g_HttpsURLConnectionGetDefaultHostnameVerifier;
+
+// javax/net/ssl/KeyManagerFactory
+jclass    g_KeyManagerFactory;
+jmethodID g_KeyManagerFactoryGetInstance;
+jmethodID g_KeyManagerFactoryInit;
+jmethodID g_KeyManagerFactoryGetKeyManagers;
+
+// javax/net/ssl/SNIHostName
+jclass    g_SNIHostName;
+jmethodID g_SNIHostNameCtor;
+
 // javax/net/ssl/SSLEngine
 jclass    g_SSLEngine;
-jmethodID g_SSLEngineSetUseClientModeMethod;
-jmethodID g_SSLEngineGetSessionMethod;
-jmethodID g_SSLEngineBeginHandshakeMethod;
-jmethodID g_SSLEngineWrapMethod;
-jmethodID g_SSLEngineUnwrapMethod;
-jmethodID g_SSLEngineCloseInboundMethod;
-jmethodID g_SSLEngineCloseOutboundMethod;
-jmethodID g_SSLEngineGetHandshakeStatusMethod;
+jmethodID g_SSLEngineGetApplicationProtocol;
+jmethodID g_SSLEngineSetUseClientMode;
+jmethodID g_SSLEngineGetSession;
+jmethodID g_SSLEngineBeginHandshake;
+jmethodID g_SSLEngineWrap;
+jmethodID g_SSLEngineUnwrap;
+jmethodID g_SSLEngineCloseOutbound;
+jmethodID g_SSLEngineGetHandshakeStatus;
+jmethodID g_SSLEngineSetSSLParameters;
 
 // java/nio/ByteBuffer
 jclass    g_ByteBuffer;
-jmethodID g_ByteBufferAllocateMethod;
-jmethodID g_ByteBufferPutMethod;
-jmethodID g_ByteBufferPut2Method;
-jmethodID g_ByteBufferPut3Method;
-jmethodID g_ByteBufferFlipMethod;
-jmethodID g_ByteBufferGetMethod;
-jmethodID g_ByteBufferPutBufferMethod;
-jmethodID g_ByteBufferLimitMethod;
-jmethodID g_ByteBufferRemainingMethod;
-jmethodID g_ByteBufferCompactMethod;
-jmethodID g_ByteBufferPositionMethod;
+jmethodID g_ByteBufferAllocate;
+jmethodID g_ByteBufferCompact;
+jmethodID g_ByteBufferFlip;
+jmethodID g_ByteBufferGet;
+jmethodID g_ByteBufferLimit;
+jmethodID g_ByteBufferPosition;
+jmethodID g_ByteBufferPutBuffer;
+jmethodID g_ByteBufferPutByteArray;
+jmethodID g_ByteBufferPutByteArrayWithLength;
+jmethodID g_ByteBufferRemaining;
```

Comment:
Would it make sense to use some of the [JNI NIO apis](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html#NewDirectByteBuffer) to make the byte buffers instead of calling the byte buffer methods through JNI?

Reply:
I stuck with the simplicity of going through these methods, since they handle all the mark / position / limit tracking. If we do the direct access through the JNI NIO support, we'd have to update those ourselves which would make things a bit more complex.

I'd prefer to go with the simpler way right now and note it as a possible future improvement/investigation that could improve perf.